localstack-core/localstack/services/kinesis/packages.py

```
@@ -1,28 +1,82 @@
 import os
+from enum import StrEnum
 from functools import lru_cache
-from typing import List
+from typing import Any, List
 
-from localstack.packages import Package
-from localstack.packages.core import NodePackageInstaller
+from localstack.packages import InstallTarget, Package
+from localstack.packages.core import GitHubReleaseInstaller, NodePackageInstaller
+from localstack.packages.java import JavaInstallerMixin, java_package
 
-_KINESIS_MOCK_VERSION = os.environ.get("KINESIS_MOCK_VERSION") or "0.4.9"
+_KINESIS_MOCK_VERSION = os.environ.get("KINESIS_MOCK_VERSION") or "0.4.12"
 
 
-class KinesisMockPackage(Package[NodePackageInstaller]):
-    def __init__(self, default_version: str = _KINESIS_MOCK_VERSION):
+class KinesisMockEngine(StrEnum):
+    NODE = "node"
+    SCALA = "scala"
+
+    @classmethod
+    def _missing_(cls, value: str | Any) -> str:
+        # default to 'node' if invalid enum
+        if not isinstance(value, str):
+            return cls(cls.NODE)
+        return cls.__members__.get(value.upper(), cls.NODE)
```

Nit: Not entirely happy with this fallback logic here - it is effectively unused, as we only ever check if it is scala anyway, and we would silently accept obviously wrong values. Also, if we ever switch the default, we have to do it in multiple places. Nothing to hold back the merge, but I think we could potentially make this easier.

---

Yeah I suppose this is unideal but I wanted to make sure (since we're feature-flagging this strategy) that nothing gets broken if incorrect values are used. Definitely erring on the defensive side.

What have we done in past for situations like this?

---

Usually we would fallback, but at least with a warning in the log.